spec.bs

```diff
@@ -2024,6 +2040,58 @@ The IDL attribute {{HTMLSharedStorageWritableElementUtils/sharedStorageWritable}
     1. Return |parameters|[|paramKey|].
   </div>
 
+# Web Locks Integration # {#web-locks-integration}
+
+## User Agent Associated State ## {#user-agent-associated-state}
+
+A <dfn>shared storage lock managers map</dfn> is a [=map=] of [=/origins=] to [=lock managers=]. It is initially empty.
```

Comment:
Shared storage defines its own [shared storage shed](https://wicg.github.io/shared-storage/#shared-storage-shed), which is a map from `origins` to `storage shelves`.

Special casing the [obtain a storage bottle map](https://storage.spec.whatwg.org/#obtain-a-storage-bottle-map) algorithm could work (i.e, if it's a Shared Storage environment, use the shelf from [shared storage shed](https://wicg.github.io/shared-storage/#shared-storage-shed)). However, as I remember, the Web Locks & Storage Buckets owners preferred separating this functionality for Shared Storage to avoid potential side effects (e.g., calling [=obtain a local storage bottle map=] may have other redundancies that Shared Storage doesn't care).

However, I'm unsure if this preference on implementation still applies at the specification level.

@inexorabletash / @ayuishii: Could you weigh in on this? (also cc'ing @evanstade, though it looks he's currently unavailable)

Reply:
Is the origin key because shared storage cannot be used from a partitioned context?  Its only ever usable from the top-level?  (I think there have been some proposals around partitioned top-level contexts, so not sure we should depend on that at the spec level.)

Reply:
Shared storage can be written in any context, but the read can only be from 'shared storage worklets', and the only way data can leave these worklets is via some privacy-preserving APIs.

Reply:
Right, I remember now.  Shared storage effectively bypasses partitioning by design and is constrained by the privacy gates.

Reply:
My memory here is hazy, so I don't remember any specific objections. Special casing "obtain a storage bottle map" seems reasonable if the Storage owners are okay with it, since it wouldn't require any changes in Web Locks itself which sounds like the right layering?

Tagging in @saschanaz too for spec thoughts.